packages/azure-services/src/azure-batch/batch.spec.ts

```diff
@@ -98,6 +116,171 @@ describe(Batch, () => {
+                        { name: 'TASK_ARGUMENTS', value: JSON.stringify(taskArguments) },
+                    ] as BatchServiceModels.EnvironmentSetting[],
+                    executionInfo: {
+                        exitCode: 1,
+                        result: 'failure',
+                        failureInfo: {
+                            category: 'userError' as ErrorCategory,
+                            code: '1',
+                            message: 'The task was ended by user request',
+                        } as TaskFailureInformation,
+                    } as TaskExecutionInformation,
+                    stateTransitionTime: timestamp,
+                },
+                {
+                    id: 'id-4: task without environment settings data',
+                    environmentSettings: undefined,
+                    executionInfo: {
+                        exitCode: 1,
+                        result: 'failure',
+                        failureInfo: {
+                            category: 'userError' as ErrorCategory,
+                            code: '1',
+                            message: 'The task was ended by user request',
+                        } as TaskFailureInformation,
+                    } as TaskExecutionInformation,
+                    stateTransitionTime: timestamp,
+                },
+                {
+                    id: 'id-5: task without TASK_ARGUMENTS environment value',
+                    environmentSettings: [{ name: 'name-1', value: 'value-1' }] as BatchServiceModels.EnvironmentSetting[],
+                    executionInfo: {
+                        exitCode: 1,
+                        result: 'failure',
+                        failureInfo: {
+                            category: 'userError' as ErrorCategory,
+                            code: '1',
+                            message: 'The task was ended by user request',
+                        } as TaskFailureInformation,
+                    } as TaskExecutionInformation,
+                    stateTransitionTime: timestamp,
+                },
+            ];
+
+            const items1 = new TaskListStub(tasks.slice(0, 2));
+            items1.odatanextLink = 'odatanextLink-1';
+            const items2 = new TaskListStub(tasks.slice(2, tasks.length));
+
+            const options = {
+                taskListOptions: { filter: `state eq 'completed' and executionInfo/result eq 'failure'` },
+            };
+            taskMock
+                .setup(async o => o.list(config.jobId, options))
+                .returns(async () => Promise.resolve(<TaskListResponse>(<unknown>items1)))
+                .verifiable();
+            taskMock
+                .setup(async o => o.listNext(items1.odatanextLink, options))
+                .returns(async () => Promise.resolve(<TaskListResponse>(<unknown>items2)))
+                .verifiable();
+
+            const expectedFailedTasks: BatchTask[] = [
```

Comment:
can you refactor this & create a helper method to generate these objects?

Reply:
No, generalization in UTs is not always the best and will not help future understanding of test data/test flow. In this case it has no high benefits either as only few properties are common in array of objects. The simple declaration is only required here as well.